src/codewhisperer/util/authUtil.ts

```diff
@@ -79,14 +78,9 @@ export class AuthUtil {
             }
         })
 
-        this.secondaryAuth.onDidChangeActiveConnection(async conn => {
-            if (conn?.type === 'sso') {
-                this.usingEnterpriseSSO = !isBuilderIdConnection(conn)
-                if (!this.isConnectionExpired() && this.usingEnterpriseSSO) {
-                    vscode.commands.executeCommand('aws.codeWhisperer.notifyNewCustomizations')
-                }
-            } else {
-                this.usingEnterpriseSSO = false
+        this.secondaryAuth.onDidChangeActiveConnection(async () => {
+            if (this.isValidEnterpriseSsoInUse()) {
```

Comment:
When `secondaryAuth` emits an event we can assume it already updated all of its state. So if we use `this.conn` it will pull the latest connection from `secondaryAuth`